docs/sources/developers/plugins/backend/_index.md

```diff
@@ -13,63 +13,80 @@ title: Backend plugins
 
 ## Use cases for implementing a backend plugin
 
-The following examples gives you an idea of why you'd consider implementing a backend plugin:
+The following examples gives are some common use cases for backend plugins:
 
 - Enable [Grafana Alerting]({{< relref "../../../alerting/" >}}) for data sources.
-- Connect to non-HTTP services that normally can't be connected to from a web browser, e.g. SQL database servers.
-- Keep state between users, e.g. query caching for data sources.
+- Connect to SQL database servers and other non-HTTP services that normally can't be connected to from a browser.
+- Keep state between users, for example, by query caching for data sources.
 - Use custom authentication methods and/or authorization checks that aren't supported in Grafana.
-- Use a custom data source request proxy, see [Resources]({{< relref "#resources" >}}).
+- Use a custom data source request proxy (refer to [Resources]({{< relref "#resources" >}}) for more information.
+
+## Grafana backend plugin system
+
+The Grafana backend plugin system is based on HashiCorp's [Go Plugin System over RPC](https://github.com/hashicorp/go-plugin). Our implementation of the Grafana server launches each backend plugin as a subprocess and communicates with it over [gRPC](https://grpc.io/).
+
+### Benefits for plugin development
 
-## Grafana’s backend plugin system
+Grafana's approach has benefits for developers:
 
-The Grafana backend plugin system is based on the [go-plugin library by HashiCorp](https://github.com/hashicorp/go-plugin). The Grafana server launches each backend plugin as a subprocess and communicates with it over [gRPC](https://grpc.io/). This approach has a number of benefits:
+- **Stability:** Plugins can’t crash your Grafana process: a panic in a plugin doesn’t panic the server.
+- **Ease of development:\*** Plugins can be written in any language that supports gRPC (for example, just write a Go application and run `go build`).
+- **Security:** Plugins only have access to the interfaces and arguments that are given to it, not to the entire memory space of the process.
 
-- Plugins can’t crash your grafana process: a panic in a plugin doesn’t panic the server.
-- Plugins are easy to develop: just write a Go application and run `go build` (or use any other language which supports gRPC).
-- Plugins can be relatively secure: The plugin only has access to the interfaces and arguments that are given to it, not to the entire memory space of the process.
+### Capabilities of the backend plugin system
 
-Grafana's backend plugin system exposes a couple of different capabilities, or building blocks, that a backend plugin can implement:
+Grafana's backend plugin system exposes several key capabilities, or building blocks, that your backend plugin can implement:
 
 - Query data
 - Resources
 - Health checks
 - Collect metrics
```

Comment:
Not related to those changes, but shall we add data streaming to the list as well?

I couldn't find any docs about streaming backend plugins, only this tutorial:

https://grafana.com/tutorials/build-a-streaming-data-source-plugin/

And this but it's only for the frontend:

https://grafana.com/docs/grafana/latest/developers/plugins/build-a-streaming-data-source-plugin/

CC @sympatheticmoose - WDYT?

Reply:
It's worth adding, but as a follow up like you say 👍 we can take a look at both those and https://github.com/grafana/grafana-plugin-examples/tree/main/examples/datasource-streaming-websocket

Reply:
I have created https://github.com/grafana/grafana/issues/68741 to track.